src/test/java/com/pmarko09/school_supervision/controller/TeacherControllerTest.java

```diff
@@ -0,0 +1,209 @@
+@AutoConfigureMockMvc
+public class TeacherControllerTest {
+
+    @MockBean
+    TeacherService teacherService;
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Test
+    void getTeachers_DataCorrect_ReturnStatus200() throws Exception {
+        //given
+        List<TeacherDTO> teacherDTOS = List.of(
+                TeacherDTO.builder()
+                        .id(1L)
+                        .firstname("Jan")
+                        .lastname("Kowal")
+                        .email("Jan@")
+                        .build(),
+                TeacherDTO.builder()
+                        .id(2L)
+                        .firstname("A")
+                        .lastname("B")
+                        .email("x@")
+                        .build()
+        );
+
+        when(teacherService.getAllTeachers()).thenReturn(teacherDTOS);
+
+        //when then
+        mockMvc.perform(MockMvcRequestBuilders.get("/teachers"))
+                .andDo(print())
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(jsonPath("$").isArray())
+                .andExpect(jsonPath("$[0].id").value(1L))
+                .andExpect(jsonPath("$[0].firstname").value("Jan"))
+                .andExpect(jsonPath("$[0].lastname").value("Kowal"))
+                .andExpect(jsonPath("$[0].email").value("Jan@"))
+                .andExpect(jsonPath("$[1].id").value(2L))
+                .andExpect(jsonPath("$[1].firstname").value("A"))
+                .andExpect(jsonPath("$[1].lastname").value("B"))
+                .andExpect(jsonPath("$[1].email").value("x@"));
+    }
+
+    @Test
+    void getTeacher_DataCorrect_ReturnStatus200() throws Exception {
+        //given
+        TeacherDTO teacherDTO = TeacherDTO.builder()
+                .id(1L)
+                .firstname("Jan")
+                .lastname("Kowal")
+                .email("Jan@")
+                .build();
+
+        when(teacherService.getTeacher(1L)).thenReturn(teacherDTO);
+
+        //when then
+        mockMvc.perform(MockMvcRequestBuilders.get("/teachers/1"))
+                .andDo(print())
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(jsonPath("$.id").value(1L))
+                .andExpect(jsonPath("$.firstname").value("Jan"))
+                .andExpect(jsonPath("$.email").value("Jan@"));
+    }
+
+    @Test
+    void addTeacher_DataCorrect_ReturnStatus200() throws Exception {
+        //given
+        Teacher teacher = new Teacher();
+        teacher.setFirstname("A");
+        teacher.setLastname("B");
+        teacher.setEmail("123@");
+        teacher.setPassword("444");
+
+        TeacherDTO teacherDTO = TeacherDTO.builder()
+                .id(1L)
+                .firstname("A")
+                .lastname("B")
+                .email("123@")
+                .build();
+//TODO: pytanie o to czemu wpisujac do addTeacher(teacher) nie przechodzilo
```

Comment:
Masz zaimplementowany equals oraz hashcode. Sprawdz czy ten teacher ktory przekazujesz z tym ktorego porownujesz to czy wywoalnie metody equals zwraca dla nich true